purchases_test.go

```diff
@@ -0,0 +1,259 @@
+					<payment_method>credit_card</payment_method>
+					<reference>8001561</reference>
+					<source>transaction</source>
+					<recurring type="boolean">false</recurring>
+					<test type="boolean">true</test>
+					<voidable type="boolean">true</voidable>
+					<refundable type="boolean">true</refundable>
+					<ip_address nil="nil"></ip_address>
+					<gateway_type>test</gateway_type>
+					<origin>api</origin>
+					<description>Transaction Description</description>
+					<message>Successful test transaction</message>
+					<approval_code nil="nil"></approval_code>
+					<failure_type nil="nil"></failure_type>
+					<gateway_error_codes nil="nil"></gateway_error_codes>
+					<cvv_result code="" nil="nil"></cvv_result>
+					<avs_result code="D">Street address and postal code match.</avs_result>
+					<avs_result_street nil="nil"></avs_result_street>
+					<avs_result_postal nil="nil"></avs_result_postal>
+					<created_at type="datetime">2018-06-19T01:13:26Z</created_at>
+					<collected_at type="datetime">2018-06-19T01:13:26Z</collected_at>
+					<updated_at type="datetime">2018-06-19T01:13:26Z</updated_at>
+					<details>
+					<account>
+						<account_code>1</account_code>
+						<first_name>Verena</first_name>
+						<last_name>Example</last_name>
+						<company nil="nil"></company>
+						<email>verena@example.com</email>
+						<billing_info type="credit_card">
+							<first_name>Verena</first_name>
+							<last_name>Example</last_name>
+							<address1>123 Main St.</address1>
+							<address2></address2>
+							<city>SF</city>
+							<state>CA</state>
+							<zip>94105</zip>
+							<country>US</country>
+							<phone></phone>
+							<vat_number></vat_number>
+							<card_type>Visa</card_type>
+							<year type="integer">2020</year>
+							<month type="integer">6</month>
+							<first_six>411111</first_six>
+							<last_four>1111</last_four>
+						</billing_info>
+					</account>
+					</details>
+					</transaction>
+				</transactions>
+				<a name="refund" href="https://your-subdomain.recurly.com/v2/invoices/2531/refund" method="post"/>
+			</charge_invoice>
+			<credit_invoices type="array">
+			</credit_invoices>
+		</invoice_collection>`)
+	})
+
+	r, invs, err := client.Purchases.Create(recurly.Purchase{})
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	} else if r.IsError() {
+		t.Fatal("expected preview subscription to return OK")
+	}
+
+	expectedInvoiceNumber := 2531
+	expectedTax := 111
+	expectedTotal := 1411
+
+	if invs.ChargeInvoice.InvoiceNumber != expectedInvoiceNumber {
+		t.Errorf("expected invoice number %d; got %d\n", expectedInvoiceNumber, invs.ChargeInvoice.InvoiceNumber)
+	}
+	if invs.ChargeInvoice.TaxInCents != expectedTax {
+		t.Errorf("expected tax %d; got %d\n", expectedTax, invs.ChargeInvoice.TaxInCents)
+	}
+	if invs.ChargeInvoice.TotalInCents != expectedTotal {
+		t.Errorf("expected total %d; got %d\n", expectedTotal, invs.ChargeInvoice.TotalInCents)
+	}
```

Comment:
`if`/`else if` with `t.Fatalf` instead of `Errorf` to be consistent with the rest of the repo. Here's an example:
https://github.com/blacklightcms/recurly/blob/master/accounts_test.go#L153-L161

Reply:
Will do!